static/js/util/profile_edit.js

```diff
@@ -334,72 +300,94 @@ export function boundMonthYearField(keySet, label) {
+    // will have text at a time since the user can't edit more than one field at once
+    // so we need to look in the state to see
+    let newTuple = Object.assign({}, tuple, {
+      year: year !== undefined ? year : tuple.year,
+      month: month !== undefined ? month : tuple.month,
+      date: date !== undefined ? date : tuple.date
     });
+    
+    // these functions return undefined if a month, date, or year is invalid, and converts the value to a number
+    // except if the input value is an empty string, in which case an empty string is returned
+    let validatedDate = 1;
+    if (!omitDate) {
+      validatedDate = validateDate(newTuple.date);
+    }
+    let validatedMonth = validateMonth(newTuple.month);
+    let validatedYear = validateYear(newTuple.year);
 
-    // these functions return undefined if month or year is invalid, and converts the value to a number
-    let validatedMonth = validateMonth(newMonthYear.month);
-    let validatedYear = validateYear(newMonthYear.year);
+    // keep text up to date
+    _.set(clone, editKeySet, newTuple);
 
-    if (validatedMonth === undefined || validatedYear === undefined) {
-      // store the edit value and make the formatted date null so it doesn't pass validation
-      _.set(clone, editKeySet, newMonthYear);
-      _.set(clone, keySet, undefined);
-    } else if ( validatedMonth === "" || validatedYear === "") {
-      _.set(clone, editKeySet, undefined);
+    if (validatedDate === undefined || validatedMonth === undefined || validatedYear === undefined) {
+      // store the edit value and make the formatted date undefined so it doesn't pass validation
       _.set(clone, keySet, null);
     } else {
-      // format date and store it, and erase the edit data
-      let formattedDate = moment().
-        set('year', validatedYear).
-        set('month', validatedMonth - 1).
-        set('date', 1).
-        format(DATE_FORMAT);
-      _.set(clone, editKeySet, undefined);
-      _.set(clone, keySet, formattedDate);
+      let momentDate = moment(`${validatedYear}-${validatedMonth}-${validatedDate}`, ISO_8601_FORMAT);
+      if (!momentDate.isValid()) {
+        // date is invalid according to moment.js so make the formatted date undefined so it
+        // doesn't pass validation
+        _.set(clone, keySet, null);
+      } else {
+        // format date and store it
+        _.set(clone, keySet, momentDate.format(ISO_8601_FORMAT));
+      }
     }
     updateProfile(clone);
   };
 
-  let monthYear = getMonthYear();
+  let tuple = getTuple();
+
+  let dateField, dateSlash;
+  if (!omitDate) {
+    dateSlash = <span className="slash"> / </span>;
+    dateField = <TextField
+      hintText="DD"
+      floatingLabelText=" "
+      floatingLabelFixed={true}
+      style={{
+        maxWidth: "2em"
+      }}
+      fullWidth={true}
+      value={tuple.date !== undefined ? tuple.date : ""}
+      onChange={e => setNewDate(e.target.value, undefined, undefined)}
+    />;
+  }
```

Comment:
I would refactor this bit to look more like:

```js
let dateSlash = () => omitDate ? undefined : <span className="slash"> / </span>;
let dateField = () => {
  if ( !omitDate ) {
    return <TextField>
      ...
    </TextField>
}
```

then in the JSX below we can just call `{dateField()}`.

idk why but this way feels cleaner to me, feel free to stick with what you have though!